src/prop_name.rs

```diff
@@ -0,0 +1,312 @@
+use crate::ffi_util::CStrLike;
+
+use std::ffi::{CStr, CString};
+
+/// A borrowed name of a RocksDB property.
+///
+/// The value is guaranteed to be a nul-terminated UTF-8 string.  This means it
+/// can be converted to [`CStr`] and [`str`] at zero cost.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct PropName(CStr);
+
+impl PropName {
+    /// Creates a new object from a nul-terminated string with no internal nul
+    /// bytes.
+    ///
+    /// Panics if the `value` isn’t terminated by a nul byte or contains
+    /// interior nul bytes.
+    pub(crate) const fn new_unwrap(value: &str) -> &Self {
+        let bytes = if let Some((&0, bytes)) = value.as_bytes().split_last() {
+            bytes
+        } else {
+            panic!("input was not nul-terminated");
+        };
+
+        let mut idx = 0;
+        while idx < bytes.len() {
+            assert!(bytes[idx] != 0, "input contained interior nul byte");
```

Comment:
```suggestion
            assert_ne!(bytes[idx], 0, "input contained interior nul byte");
```

Reply:
Alas:

    error: `Arguments::<'a>::new_v1` is not yet stable as a const fn
    error[E0015]: cannot call non-const fn `assert_failed::<u8, u8>` in constant functions

Reply:
Yeah. I see. See comment below.

Reply:
It seems this assert doesn't work as well.

Reply:
Should be something like this: 
```rust
       let bytes = value.as_bytes();

        if !(bytes.len() > 1 && bytes[bytes.len() - 1] == 0) {
            panic!("input was not nul-terminated");
        }

        let mut idx = 0;
        while idx < bytes.len() - 1 {
            if bytes[idx] == 0 {
                panic!("input contained interior nul byte");
            }

            idx += 1;
        }
```

Reply:
It should work fine.  Not sure when, but `assert` without formatting is const:

    $ c +1.60 t --lib -- prop_name
        Finished test [unoptimized] target(s) in 0.02s
         Running unittests (target/debug/deps/rocksdb-86da67792af28854)

    running 3 tests
    test prop_name::sanity_checks ... ok
    test prop_name::test_non_nul_terminated - should panic ... ok
    test prop_name::test_interior_nul - should panic ... ok

    test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 10 filtered out; finished in 0.00s

1.60 is Rust version specified in Cargo.toml